src/routes/home/components/DisciplineCard.tsx

```diff
@@ -37,16 +40,35 @@ interface PropsType {
   discipline: Discipline
   showProgress: boolean
   onPress: (profession: Discipline) => void
-  navigateToNextExercise: (profession: Discipline) => void
+  navigateToNextExercise: (
+    disciplineId: number,
+    exerciseKey: number,
+    disciplineTitle: string,
+    documents: Document[]
+  ) => void
 }
 
 const DisciplineCard = (props: PropsType): ReactElement => {
   const { discipline, showProgress, onPress, navigateToNextExercise } = props
   const { data: progress } = useReadProgress(discipline)
   const moduleAlreadyStarted = progress !== null && progress !== 0
+  const [documents, setDocuments] = useState<Document[] | null>(null)
+  const { data: nextExercise } = useReadNextExercise(discipline)
+
+  useEffect(() => {
+    if (nextExercise) {
+      loadDocuments({ disciplineId: nextExercise.disciplineId })
+        .then(data => {
+          setDocuments(data)
+        })
+        .catch(reportError)
+    }
+  }, [nextExercise])
 
   const navigate = () => {
-    navigateToNextExercise(discipline)
+    if (documents !== null && nextExercise !== null) {
+      navigateToNextExercise(nextExercise.disciplineId, nextExercise.exerciseKey, '', documents) // TODO set discipline title correct LUN-320
```

Comment:
why do we need to pass the `disciplineTitle` here?
I see that according to design we need more information about the next exercise for extending DisciplineCard but i think that comes later with a separate task
https://app.zeplin.io/project/5fc7c98a435298126bcfd4ee/screen/624ac2cc6aacf3112ea90218

Reply:
Resolved in https://github.com/digitalfabrik/lunes-app/pull/290